### ISSUE TITLE
Logical assignment operators; Fix parsing of optional chaining

### ISSUE THREAD

**Issue by gilad905** [beautify-web/js-beautify#1888]
1. Add support for [logical OR/AND/nullish assignment operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators#assignment_operators)
2. Fix incorrect parsing of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) when followed by a bracket notation.

# Input - expected output should not change
```
  let a;
  a ||= 1;
  a &&= 2;
  a ??= 3;
  const b = a?.[keyName];
```

# Actual (not expected) output
```
  let a;
  a || = 1;
  a && = 2;
  a ?? = 3;
  const b = a?. [keyName];
```

P.S.
I've opened a PR for the logical operators support. It failed one of the automated tests, although if you look at the commit it's clear the commit didn't trigger the fail because the changes are very minor.

Thanks!

**Comment by gilad905** [comment]
I realized now that the beutifying of #2 still results in valid code. But I would still suggest to change it because I believe it is less readable.

**Comment by bitwiseman** [comment]
Javascript doesn't contain a `||=` operator. 
https://262.ecma-international.org/5.1/#sec-7.7 

I assume that this es6, much like `?.`.  

Note, `const b = a?.[keyName];` remains unchanged on https://beautifier.io . Are you using an editor plugin?

**Comment by gilad905** [comment]
yes I'm using [atom-beautify](https://github.com/Glavin001/atom-beautify)

**Comment by mhnaeem** [comment]
Fixed in [#1991](https://github.com/beautify-web/js-beautify/pull/1992)
